### PR TITLE
Update base image version for Agent/Cluster-Agent/dogstatsd containers

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

* Use alpine:3.17 for dogstastd image.
* Use ubuntu:23.04 for Agent and Cluster-Agent images.

### Motivation

Use the latest ubuntu version for agent and cluster-agent image

### Describe how to test/QA your changes

* Deploy the Agent/Cluster-Agent with the helm-chart or Operator and check that everything works.
* also try to install a `integrations-extra` integration to check that it still work.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: Dockerfiles/agent/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,7 @@
-FROM ubuntu:22.04 AS baseimage
+ARG BASE_IMAGE_UBUNTU_VERSION=23.04
+FROM ubuntu:$BASE_IMAGE_UBUNTU_VERSION AS baseimage
 LABEL baseimage.os "ubuntu jammy LTS"
```
Comment (sblumenthal) [inline Dockerfiles/agent/Dockerfile:3]: Should this also be changed to `lunar`?

File: Dockerfiles/cluster-agent/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 # Preparation stage: layout and chmods #
 ########################################
-
-FROM ubuntu:22.04 as builder
+ARG BASE_IMAGE_UBUNTU_VERSION=23.04
+FROM ubuntu:$BASE_IMAGE_UBUNTU_VERSION as builder
 ARG TARGETARCH
-LABEL baseimage.os "ubuntu jammy LTS"
-LABEL baseimage.name "ubuntu:22.04"
+LABEL baseimage.os "ubuntu lunar"
```
Comment (sblumenthal) [inline Dockerfiles/cluster-agent/Dockerfile:7]: Do we want to add this string as an ARG as well?

File: Dockerfiles/agent/Dockerfile (hunk 1 above)
Comment (clamoriniere) [reply Dockerfiles/agent/Dockerfile:3]: yes indeed

File: Dockerfiles/cluster-agent/Dockerfile (hunk 2 above)
Comment (clamoriniere) [reply Dockerfiles/cluster-agent/Dockerfile:7]: yes, 
the idea in futur PR is to use the `release.json` to store the base info image, and provide this info to the `invoke build` command

File: Dockerfiles/cluster-agent/Dockerfile [hunk 3]
```diff
@@ -37,9 +38,11 @@ RUN gcc -pipe -Wall -Wextra -O2 -shared -fPIC -Wl,--version-script=/tmp/nosys.sy
 # Actual docker image construction #
 ####################################
 
-FROM ubuntu:22.04
+FROM ubuntu:$BASE_IMAGE_UBUNTU_VERSION
 
 LABEL maintainer "Datadog <package@datadoghq.com>"
+LABEL baseimage.os "ubuntu lunar"
```
Comment (sblumenthal) [inline Dockerfiles/cluster-agent/Dockerfile:44]:
```suggestion
LABEL baseimage.os "ubuntu ${BASE_IMAGE_UBUNTU_NAME}"
```

**Review by sblumenthal — APPROVED** [review]

File: releasenotes-dca/notes/update-cluster-agent-container-base-image-to-Ubuntu-23.04-lunar-015b473b575a61b1.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+enhancements:
+  - |
+    The Datadog cluster-agent container image is now using Ubuntu 23.04 lunar as base image.
```
Comment (rtrieu) [inline releasenotes-dca/notes/update-cluster-agent-container-base-image-to-Ubuntu-23.04-lunar-015b473b575a61b1.yaml:4]:
```suggestion
    The Datadog cluster-agent container image is now using Ubuntu 23.04 lunar as the base image.
```

File: releasenotes/notes/update-agent-container-base-image-to-Ubuntu-23.04-lunar-015b473b575a61b1.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+enhancements:
+  - |
+    The Datadog agent container image is now using Ubuntu 23.04 lunar as base image.
```
Comment (rtrieu) [inline releasenotes/notes/update-agent-container-base-image-to-Ubuntu-23.04-lunar-015b473b575a61b1.yaml:4]:
```suggestion
    The Datadog agent container image is now using Ubuntu 23.04 lunar as the base image.
```

**Review by rtrieu — APPROVED** [review]
Approving with minor suggestions!

**Pull request merged**